### PR TITLE
Add jam-auth code to exp-addons

### PR DESCRIPTION
**Pull request by samanehsan**
Add jam-auth component, authenticator & authorizer to exp-addons since it is used by both lookit and ISP

### CODE REVIEW

File: exp-models/addon/authenticators/jam-jwt.js [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+
+import { jwt_decode } from 'ember-cli-jwt-decode';
+
+import moment from 'moment';
+
+const {
+    $, RSVP
+} = Ember;
+
+export default BaseAuthenticator.extend({
```
Comment (samchrisinger) [inline exp-models/addon/authenticators/jam-jwt.js]: @samanehsan can you please reformat this code so it matches with https://github.com/CenterForOpenScience/exp-addons/blob/develop/exp-models/.editorconfig and the other code in this repo?

File: exp-player/addon/components/jam-auth/component.js [hunk 2]
```diff
@@ -0,0 +1,25 @@
+import Ember from 'ember';
+import ENV from 'isp/config/environment';
```
Comment (samchrisinger) [inline exp-player/addon/components/jam-auth/component.js]: I think you'll need to use the `ember-get-config` addon here

File: exp-player/addon/components/jam-auth/template.hbs [hunk 3]
```diff
@@ -0,0 +1,20 @@
+<div class="well">
+    <p>Enter your <strong>Study ID</strong> and <strong>Participant ID</strong> into the fields below. Please {{link-to 'contact us' 'contact'}} if you have any questions.</p>
```
Comment (samchrisinger) [inline exp-player/addon/components/jam-auth/template.hbs]: I would prefer to see the ISP-specific representation of this component included in that repo instead. 

Can you please replace this with a more generic version? The ISP version can create a subclass of this component and override `layout` to have this template.

**Comment by samanehsan** [comment]
Done making those changes @samchrisinger!

**Pull request merged**